cmd/vela-worker/server.go

```diff
@@ -35,25 +37,31 @@ func (w *Worker) server() error {
 	// log a message indicating the start of serving traffic
 	//
 	// https://pkg.go.dev/github.com/sirupsen/logrus?tab=doc#Tracef
-	logrus.Tracef("serving traffic on %s", w.Config.API.Port)
+	logrus.Tracef("serving traffic on %s", w.Config.API.Address.Port())
 
-	// if provided, start serving traffic with TLS on the provided worker port
-	//
-	// https://pkg.go.dev/github.com/gin-gonic/gin?tab=doc#Engine.RunTLS
-	if len(w.Config.Certificate.Cert) > 0 {
-		// validate cert files exists at the provided paths
-		_, err := os.Stat(w.Config.Certificate.Cert)
-		if err != nil {
-			logrus.Fatalf("Expecting certificate file at %s, got %v", w.Config.Certificate.Cert, err)
-		}
-		_, err = os.Stat(w.Config.Certificate.Key)
-		if err != nil {
-			logrus.Fatalf("Expecting certificate key at %s, got %v", w.Config.Certificate.Key, err)
+	// if running with HTTPS, check certs are provided and run with TLS.
+	if strings.EqualFold(w.Config.API.Address.Scheme, "https") {
+		if len(w.Config.Certificate.Cert) > 0 && len(w.Config.Certificate.Key) > 0 {
+			// check that the certificate and key are both populated
+			_, err := os.Stat(w.Config.Certificate.Cert)
+			if err != nil {
+				logrus.Fatalf("Expecting certificate file at %s, got %v", w.Config.Certificate.Cert, err)
+			}
+			_, err = os.Stat(w.Config.Certificate.Key)
+			if err != nil {
+				logrus.Fatalf("Expecting certificate key at %s, got %v", w.Config.Certificate.Key, err)
+			}
+		} else {
+			logrus.Fatal("Unable to run with TLS: No certificate provided")
```

Comment:
Thoughts on potentially un-capitalizing the error messages here?

I don't think we have a clear defined format but in general, I believe we try using lowercase whenever possible 👍 

As an example:

```
# I think we want to do thiis
fmt.Errorf("sample lowercase error message")
logrus.Fatal("sample lowercase log message")

# I don't think we want to do this
fmt.Errorf("Sample capitalized error message")
logrus.Fatal("Sample capitalized log message")
```